cmd/traffic/cmd/manager/internal/mutator/watcher.go

```diff
@@ -431,23 +497,59 @@ func (c *configWatcher) Store(ctx context.Context, ac *agentconfig.Sidecar, upda
 			},
 			ObjectMeta: meta.ObjectMeta{
 				Name:      agentconfig.ConfigMap,
-				Namespace: ns,
-			},
-			Data: map[string]string{
-				ac.AgentName: yml,
+				Namespace: c.namespace,
 			},
 		}
-		dlog.Debugf(ctx, "Creating new ConfigMap %s.%s with %s", agentconfig.ConfigMap, ns, ac.AgentName)
-		_, err = api.Create(ctx, cm, meta.CreateOptions{})
-	} else {
-		if _, ok := cm.Data[ac.AgentName]; ok {
-			dlog.Debugf(ctx, "Updating %s in ConfigMap %s.%s", ac.AgentName, agentconfig.ConfigMap, ns)
-		} else {
-			dlog.Debugf(ctx, "Adding %s to ConfigMap %s.%s", ac.AgentName, agentconfig.ConfigMap, ns)
+	}
+
+	if cm.Data == nil {
+		cm.Data = make(map[string]string)
+	}
```

Comment:
The extra initialization is unnecessary because `cm.Data` is only read until it is unconditionally assigned att lines 544 and 549. It's OK to read from a `nil` map.